sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -270,11 +255,140 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
     keys. Enables resolution of Key Vault references in configuration settings.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, **kwargs) -> None:
         self._dict: Dict[str, str] = {}
         self._trim_prefixes: List[str] = []
         self._client: Optional[AzureAppConfigurationClient] = None
         self._secret_clients: Dict[str, SecretClient] = {}
+        self._key_vault_options: Optional[AzureAppConfigurationKeyVaultOptions] = kwargs.pop("key_vault_options", None)
```

Comment:
These should be keyword-only arguments rather than `**kwargs`. This applies to anything that you `pop` of `kwargs` and don't pass along. With an exception for the implementation of `@overload`ed methods.

Reply:
I might be misunderstanding something. This classes is only created from the load method, where all these came in, and are `@overload`.

Reply:
I am good with making this change, but I don't think it should be part of this PR. Given the current plan is to release a beta next I think we could do it between now and then.